### PR TITLE
[proto] NEON kswv mate-rescue — correctness + perf harness

### PR DESCRIPTION
**Pull request by nh13**
**Status: ready for review** (still draft until a second set of eyes signs off).

Enables the batched mate-rescue SW path on arm64 via the existing (previously unreachable) NEON `kswv::getScores8` kernel. Four kernel bugs found and fixed along the way. See commit `e101d37` for the detailed writeup.

## Results

**Correctness** — NEON batched vs scalar reference, byte-identical sorted SAM:
- Local macOS arm64, 500k dwgsim/chr17, 3 seeds (42, 7, 123): all 0 diffs
- Linux arm64 CI, 500k dwgsim/chr17: PASS (cross-arch identity gate)

**kswv selftest** (test/kswv_selftest.cpp, 10014 pairs, random + edge cases):
- score mismatches: 0
- coord (tb/qb) mismatches: 0
- score2 (suboptimal) mismatches: 0

**Performance** (median of 3 reps):
- Local M-series, 8 threads, 500k PE 100bp dwgsim on chr17: **1.42× (-29.4% wall time)**
- Linux arm64 CI, 4 threads, 500k PE 100bp: 1.04× (slower runner; mate rescue dilutes at lower core count)

## What changed

- `src/macro.h` — `BWAMEM_BATCHED_MATESW` macro (replaces `__AVX512BW__` gate).
- `src/bwamem.cpp:1255` — dispatcher uses the new macro.
- `src/bwamem_pair.cpp:657, 706` — inner `pwsw->getScores8/16` gates use the new macro; fatal error branch replaced with a clearer message.
- `src/kswv.cpp` — four NEON kernel fixes (te width, freeze mask, per-pair score2 exclusion, minsc filter).
- `test/kswv_selftest.cpp` — self-consistency harness, production-matching h0, tight gate.
- `Makefile` — `kswv_selftest` target, `DISABLE_BATCHED_MATESW` control-build flag.
- `.github/workflows/proto-neon-kswv.yml` — ARM selftest + port/control perf A/B + cross-arch SAM identity diff, all strict gates.

## Merge plan

PR is draft. Suggested next steps before un-drafting:
1. One reviewer scan through the four kernel fixes in `src/kswv.cpp`.
2. (Optional) Confirm speedup on c7g Graviton3 via self-hosted runner if we want a "production-class" perf number vs the free CI runner.
3. Squash / re-author the noisier scaffolding commits into a clean history before merging to `fg-main`.

## The four bugs, in order

Each was masked by the previous one, so they surfaced only as the kernel got closer to correct:

1. `te_vec` tracked 16 pairs in 8 int16 slots (hacky duplicate store) → pairs 8–15 got copies of the wrong pair's te.
2. No per-lane freeze at `KSW_XSTOP` → once a pair hit the target score, untouched ref bytes in later rows kept inflating `gmax/te`, breaking the phase-1 write-back gate for `tb/qb`.
3. score2 scan used a single aggregate `[minh+1, limit)` range across the whole batch → one pair's primary region contaminated another's suboptimal.
4. score2 scan didn't enforce `minsc` → sub-threshold plateau scores from padding decay leaked in as false suboptimals, collapsing MAPQ on real reads.

Selftest caught each of these cleanly once the right fields were checked. Iteration worked as intended.

## Summary by CodeRabbit

* **New Features**
  * CI performance self-test workflow that benchmarks ARM vs x86, verifies SAM identity, and uploads artifacts.
  * Deterministic SIMD-vs-scalar self-test executable with a Make target; builds now use C++14.
* **Bug Fixes**
  * Fixed NEON alignment scoring to prevent inter-lane interference and improved secondary-score selection.
* **Chores**
  * Added build-time toggle to choose batched vs scalar mate-rescue and updated clean/test targets.

### CODE REVIEW

File: test/kswv_selftest.cpp [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+        pairArray[pos++] = sp;
+    }
+
+    // Phase 1: reverse pass → fills tb, qb by subtraction inside the kernel.
+    // For diagnosis, save pre-phase-1 kswr_t so we can see whether phase 1's
+    // score differs from the stored phase-0 score (that equality gates the
+    // tb/qb write-back at kswv.cpp:535).
+    std::vector<kswr_t> pre_phase1(out.aln.begin(), out.aln.begin() + n);
+    pwsw->getScores8(pairArray.data(), seqBufRef.data(), seqBufQer.data(),
+                     out.aln.data(), pos, 1, 1);
```
[the remark this answers is not in the document]
Comment (nh13) [reply test/kswv_selftest.cpp:240]: The kernel writes back via `p[l].regid` (see `src/kswv.cpp:593,595`), not the compacted lane index. After compaction, `pairArray[j].regid` is preserved as the original pair index, so phase-1 still writes to `out.aln[original_idx]` — which is what `main()` then diffs. Production uses the same layout: `src/bwamem_pair.cpp:700-701` passes the full-size `aln` buffer alongside the compacted `pairArray` with count `pos8`/`pos16`. Passing a per-survivor `phase1_aln` (size `pos`) would index out-of-bounds on `regid >= pos`. Empirically: 10014/10014 pairs match scalar on score, coord, and score2. Leaving the thread open for maintainer review.

**Comment by nh13** [comment]
@coderabbitai resume

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review
